File: Example/RMessageTests/RMessageTests.m
//
//  RMessageTests.m
//  RMessageTests
//
//  Created by Adonis Peralta on 12/4/17.
//

#import "RMessage.h"
#import <XCTest/XCTest.h>

@interface RMessageTests : XCTestCase

@end

@implementation RMessageTests
{
  UIApplication *app;
}

- (void)setUp
{
  [super setUp];
  app = [UIApplication sharedApplication];
}

- (void)tearDown
{
  app = nil;
  [super tearDown];
}

- (void)testMessageInit
{
  XCTAssertNotNil([RMessage sharedMessage]);
}

- (void)testQueuedMessages
{
  for (int i = 0; i < 6; i++) {
    [RMessage showNotificationWithTitle:@"title" type:RMessageTypeError customTypeName:nil callback:nil];
  }
  NSArray *queuedMessages = [RMessage queuedMessages];
  XCTAssert(queuedMessages.count == 6);
}

@end

File: Tests/RMessageTests.swift
//
//  RMessageTests.swift
//  RMessageTests
//
//  Created by Adonis Peralta on 8/2/18.
//  Copyright © 2018 None. All rights reserved.
//

@testable import RMessage
import XCTest

class RMessageTests: XCTestCase {
  let rControl = RMController()

  override func setUp() {
    super.setUp()
  }

  override func tearDown() {
    super.tearDown()
  }

  /// Test that RMessage does not attempt to present if there is no window hierarchy present.
  func testQueuedMessages() {
    for _ in 1 ... 6 {
      rControl.showMessage(withSpec: DefaultRMessageSpec(), title: "Test")
    }
    XCTAssertTrue(rControl.queueCount == 0)
  }
}
